src/index.ts

```diff
@@ -2,8 +2,61 @@
  * Contains request routing logic.
  */
 
+import {HeaderKeys} from './headers';
+import * as injectorsCache from './injectors-cache';
 import router from './router';
 
+/** */
+async function cacheAfterServing(
+  event: FetchEvent,
+  response: Response
+): Promise<void> {
+  const xCacheAfterServing = response.headers.get(
+    HeaderKeys.X_CACHE_AFTER_SERVING
+  );
+  if (!xCacheAfterServing) {
+    return;
+  }
+
+  // Strip this internal header from the response being cached, or risk an
+  // infinite loop of saving to cache.
+  response.headers.delete(HeaderKeys.X_CACHE_AFTER_SERVING);
+
+  const [url, ...cacheKeyParts] = xCacheAfterServing.split(';');
```

Comment:
If only we had [`splitN`](https://github.com/tc39/proposal-reversible-string-split).

Reply:
I actually spent 2 hours trying to understand why the cache is being ignored because I assumed `split` works the way it does in Java/Python 🙊